packages/grafana-ui/src/components/FileDropzone/FileListItem.tsx

```diff
@@ -0,0 +1,124 @@
+import { css } from '@emotion/css';
+import { formattedValueToString, getValueFormat, GrafanaTheme2 } from '@grafana/data';
+import React from 'react';
+import { useStyles2 } from '../../themes';
+import { trimFileName } from '../../utils/file';
+import { Button } from '../Button';
+import { Icon } from '../Icon/Icon';
+import { IconButton } from '../IconButton/IconButton';
+import { DropzoneFile } from './FileDropzone';
+
+export interface FileListItemProps {
+  file: DropzoneFile;
+  removeFile?: (file: DropzoneFile) => void;
+}
+
+export function FileListItem({ file: customFile, removeFile }: FileListItemProps) {
+  const styles = useStyles2(getStyles);
+  const { file, progress, error, abortUpload, retryUpload } = customFile;
+
+  const renderRightSide = () => {
+    if (error) {
+      return (
+        <>
+          <span className={styles.error}>{error.message}</span>
+          {retryUpload ? (
+            <IconButton aria-label="Retry" name="sync" tooltip="Retry" tooltipPlacement="top" onClick={retryUpload} />
+          ) : null}
+          {removeFile ? (
+            <IconButton
+              className={retryUpload ? styles.marginLeft : ''}
+              name="trash-alt"
+              onClick={() => removeFile(customFile)}
+              tooltip="Remove"
+              aria-label="Remove"
+            />
+          ) : null}
+        </>
+      );
+    }
+
+    if (progress && file.size > progress) {
+      return (
+        <>
+          <progress className={styles.progressBar} max={file.size} value={progress} />
+          <span className={styles.paddingLeft}>{Math.round((progress / file.size) * 100)}%</span>
+          {abortUpload ? (
+            <Button variant="secondary" type="button" fill="text" onClick={abortUpload}>
+              Cancel
+            </Button>
+          ) : null}
+        </>
+      );
+    }
+    return removeFile ? (
+      <IconButton
+        name="trash-alt"
+        onClick={() => removeFile(customFile)}
+        tooltip="Remove"
+        aria-label="Remove"
```

Comment:
```suggestion
        tooltip="Remove file"
        aria-label="Remove file"
```
Just to be clear what should be removed.